### PR TITLE
add routing to 'add comments' link in slack push notifs

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @andrewconner 
https://app.asana.com/0/83234711497124/84792773514350

if the user has the extension, route to keep.url with ext open. else, route to the keep page with appropriate upsell params.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html [hunk 1]
```diff
@@ -2,11 +2,10 @@
 @import play.utils.UriEncoding
 @import com.keepit.controllers.website.DeepLinkRedirect
 
-@(r: DeepLinkRedirect, deepLinkData: JsObject)
-@deepLinkStr = @{UriEncoding.encodePathSegment(deepLinkData.toString(), "ascii")}
+@(r: DeepLinkRedirect, deepLinkData: Option[JsObject])
+@deepLinkStr = @{deepLinkData.map(data => UriEncoding.encodePathSegment(data.toString(), "ascii")).getOrElse(JsObject(Seq.empty)) }
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html]: @ryanpbrewster @jaredpetker @DerekBurch do you guys know if we need `deepLinkData`? I think the behavior may be fully specified by `r.externalLocator` here, but not sure.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala [hunk 2]
```diff
@@ -80,7 +80,7 @@ class PathCommander @Inject() (
    * KEEP
    */
   private def keepPageViaSlack(keep: Keep, slackTeamId: SlackTeamId): Path = {
-    Path(s"s/${slackTeamId.value}/k/${Keep.publicId(keep.id.get).id}/${UrlHash.hashUrl(keep.url).urlEncoded}")
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala:83]: Talk to @leogrim about removing the url hash before merging this

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala:83]: I told him to, reasons to keep it?

File: kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html (hunk 1 above)
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html]: Pretty sure that mobile needs to know that it is a message thread (which is contained in `deepLinkData`, as a `Json.obj("t" -> "m")`). Not sure though.

Comment (jaredpetker) [inline kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html]: Unsure of the full context, but mobile does not use the locator style of things ever really (and shouldn't). There is no internal router on mobile, so anytime we want to route to somewhere on mobile we should be using deep link data via json.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html]: Yeah, there's a data format for mobile. The extension uses a different format

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html]: So: this shouldn't be optional, unless you want mobile to open the home screen and be confused.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala (hunk 2 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala:83]: The `urlHash` is here to recover from the case where the keep has been deleted, and the intent is to send people to the _page_ and show the keep if possible. No need to include it when the intent is to send people to the _keep / discussion_, whether site or ext. We should probably have two different endpoints to make these intents clear.

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala:83]: There are two endpoints (`/kifi` and `/web`) already, but good point about links directly to the 3rd party content.

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala:83]: right. in this case, they are clicking "add comment" and they should end up on our page no matter what (unless we want fallback behavior, which could be confusing given their intention to "add comment"). I think for `/web` it makes sense.

File: kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/views/mobile/deepLinkRedirect.scala.html]: got it, thanks guys. I'll route them per other messages.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala (hunk 2 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PathCommander.scala:83]: :+1:

**Pull request merged**